### PR TITLE
deps: Update openshift dependencies to 4.16

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Openshift repositories do not use tags, so we update to latest commit in the `release-4.16` branch.

**Release note**:
```release-note
None
```

### CODE REVIEW

**Comment by jcanocan** [comment]
Thanks @akrejcir!
/lgtm

File: api/go.mod [hunk 1]
```diff
@@ -3,15 +3,15 @@ module kubevirt.io/ssp-operator/api
 go 1.21
 
 require (
-	github.com/openshift/api v0.0.0-20231130063041-be20875a1373 // release-4.15
-	k8s.io/apimachinery v0.28.4
+	github.com/openshift/api v0.0.0-20240419172957-f39cf2ef93fd // release-4.16
+	k8s.io/apimachinery v0.29.0
```
Comment (0xFelix) [inline api/go.mod:7]: Use `v0.29.4`?

**Review by 0xFelix — APPROVED** [review]
/approve
/lgtm

**Pull request merged**